sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -205,17 +209,17 @@ private[sql] trait DataSourceScanExec extends LeafExecNode {
 /** Physical plan node for scanning data from a relation. */
 private[sql] case class RowDataSourceScanExec(
     output: Seq[Attribute],
-    rdd: RDD[InternalRow],
+    @transient buildScan: () => RDD[InternalRow],
     @transient relation: BaseRelation,
     override val outputPartitioning: Partitioning,
     override val metadata: Map[String, String],
     override val metastoreTableIdentifier: Option[TableIdentifier])
-  extends DataSourceScanExec with CodegenSupport {
+  extends DataSourceScanExec {
 
   private[sql] override lazy val metrics =
     Map("numOutputRows" -> SQLMetrics.createMetric(sparkContext, "number of output rows"))
 
-  val outputUnsafeRows = relation match {
+  private val outputUnsafeRows = relation match {
```

Comment:
TODO: why do we need to output unsafe rows? It's weird since this operator has to inspect the relation implementation to decide whether to do this.